src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -1,957 +1,1024 @@
+
+                if (totalRecords == 0)
+                {
+                    return new PagedResult<EntityBasic>(0, 0, 0);
+                }
+
+                var pagedResult = new PagedResult<EntityBasic>(totalRecords, pageNumber, pageSize)
+                {
+                    Items = entities.Select(Mapper.Map<EntityBasic>)
+                };
+
+                return pagedResult;
+            }
+
+            //now we need to convert the unknown ones
+            switch (type)
+            {
+                case UmbracoEntityTypes.PropertyType:
+                case UmbracoEntityTypes.PropertyGroup:
+                case UmbracoEntityTypes.Domain:
+                case UmbracoEntityTypes.Language:
+                case UmbracoEntityTypes.User:
+                case UmbracoEntityTypes.Macro:
+                default:
+                    throw new NotSupportedException("The " + typeof(EntityController) + " does not currently support data for the type " + type);
+            }
+        }
+
+        [Obsolete("This method is obsolete, use the overload with dataTypeId instead", false)]
+        public PagedResult<EntityBasic> GetPagedDescendants(
+            int id,
+            UmbracoEntityTypes type,
+            int pageNumber,
+            int pageSize,
+            string orderBy = "SortOrder",
+            Direction orderDirection = Direction.Ascending,
+            string filter = "")
+        {
+            return GetPagedDescendants(id, type, pageNumber, pageSize,
+                null, orderBy, orderDirection, filter);
+        }
+
+        public PagedResult<EntityBasic> GetPagedDescendants(
+            int id,
+            UmbracoEntityTypes type,
+            int pageNumber,
+            int pageSize,
+            Guid? dataTypeId,
+            string orderBy = "SortOrder",
+            Direction orderDirection = Direction.Ascending,
+            string filter = "")
+        {
+            if (pageNumber <= 0)
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+            if (pageSize <= 0)
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+
+            var objectType = ConvertToObjectType(type);
+            if (objectType.HasValue)
+            {
+                IEnumerable<IUmbracoEntity> entities;
+                long totalRecords;
+
+                if (id == Constants.System.Root)
+                {
+                    // root is special: we reduce it to start nodes
+
+                    int[] aids = null;
+                    switch (type)
+                    {
+                        case UmbracoEntityTypes.Document:
+                            aids = Security.CurrentUser.CalculateContentStartNodeIds(Services.EntityService);
+                            break;
+                        case UmbracoEntityTypes.Media:
+                            aids = Security.CurrentUser.CalculateMediaStartNodeIds(Services.EntityService);
+             
```

Comment:
I think we should just make `Guid? dataTypeId = null` as an optional parameter - webapi might have issues selecting either of these two actions (not sure, but can easily just be one)